www/js/survey/enketo/enketo-add-note-button.js

```diff
@@ -0,0 +1,287 @@
+/*
+ * Directive to display a survey to add notes to a trip or place
+ * 
+ * Assumptions:
+ * - The directive is embedded within an ion-view
+ * - The controller for the ion-view has a function called
+ *      'recomputeDisplayTrips` which modifies the trip *list* as necessary. An
+ *      example with the label view is removing the labeled trips from the
+ *      "toLabel" filter. Function can be a no-op (for example, in
+ *      the diary view)
+ * - The view is associated with a state which we can record in the client stats.
+ * - The directive implements a `verifyTrip` function that can be invoked by
+ *      other components.
+ */
+
+angular.module('emission.survey.enketo.add-note-button',
+    ['emission.stats.clientstats',
+        'emission.services',
+        'emission.config.dynamic',
+        'emission.survey.enketo.launch',
+        'emission.survey.enketo.answer',
+        'emission.survey.enketo.preview',
+        'emission.survey.inputmatcher'])
+.directive('enketoAddNoteButton', function() {
+  return {
+    scope: {
+      trip: '=',
+      notesConfig: '=',
+      surveys: '=',
+      timeBounds: '=',
+      datakey: '@',
+    },
+    controller: "EnketoAddNoteButtonCtrl",
+    templateUrl: 'templates/survey/enketo/add-note-button.html'
+  };
+})
+.controller("EnketoAddNoteButtonCtrl", function($scope, $element, $attrs, $translate,
+    EnketoSurveyLaunch, $ionicPopover, ClientStats, DynamicConfig,
+    EnketoNotesButtonService) {
+  console.log("Invoked enketo directive controller for add-note-button");
+  $scope.notes = [];
+
+  EnketoNotesButtonService.key = $scope.datakey;
+
+  $scope.displayLabel = () => {
+    const localeCode = $translate.use();
+    // if already filled in
+    //   return $scope.notesConfig?.['filled-in-label']?.[localeCode];
+    return $scope.notesConfig?.['not-filled-in-label']?.[localeCode];
+  }
+
+  $scope.getPartialTimeUseResponse = () => {
+    const timebounds = $scope.timeBounds(); // for trips, these times are start and end of trip
+                                            // for places, it is enter and exit time (end of one trip and start of next)
+
+    // Prefill the time bounds of the trip into the survey
+    const startDayAndTime = timebounds.start_fmt_time.split('T');
+    const startHMS = startDayAndTime[1].substring(0, 8); // truncated to 8 chars for HH:MM:SS
+    const startTimezoneCode = startDayAndTime[1].split('-')[1];
+    const startTime = `${startHMS}.000-${startTimezoneCode}`
+    const endDayAndTime = timebounds.end_fmt_time.split('T');
+    const endHMS = endDayAndTime[1].substring(0, 8); // truncated to 8 chars for HH:MM:SS
+    const endTimezoneCode = endDayAndTime[1].split('-')[1];
+    const endTime = `${endHMS}.000-${endTimezoneCode}`
```

Comment:
at the very least, we should change this to use moment instead of this hardcoded parsing. But let's see if we can do anything more elegant as well.